### PR TITLE
[12.0] [MIG] purchase_discount

### PR DESCRIPTION
**Pull request by lmiguens-solvos**
Supersedes #668

### CODE REVIEW

File: purchase_discount/views/product_supplierinfo_view.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+                  <field name="discount"/>
+              </field>
+         </field>
+     </record>
+
+    <record model="ir.ui.view" id="product_supplierinfo_tree_view">
+         <field name="model">product.supplierinfo</field>
+         <field name="inherit_id" ref="product.product_supplierinfo_tree_view"/>
+         <field name="arch" type="xml">
+              <field name="price" position="after">
```
Comment (dalonsod) [inline purchase_discount/views/product_supplierinfo_view.xml:17]: Discount field in product supplier info form isn't properly shown. You should use instead

```suggestion
              <label for="date_start" position="before">
                   <label for="discount"/>
                   <div>
                      <field name="discount" class="oe_inline"/>
                  </div>              
              </label>
```

**Review by dalonsod — CHANGES_REQUESTED** [review]

File: purchase_discount/views/product_supplierinfo_view.xml (hunk 1 above)
Comment (lmiguens-solvos) [reply purchase_discount/views/product_supplierinfo_view.xml:17]: Done!!!

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Hi @lmiguens-solvos. 

Thanks a lot for porting this module. 
I began a talk here to simplify modules, merging ``purchase_discount`` and ``purchase_supplierinfo_discount`` in one module.

I will do a PR against your branch today if you're ok.

**Comment by dalonsod** [comment]
@legalsylvain that's already done in commit [82536a4](https://github.com/OCA/purchase-workflow/commit/82536a40eb1064da39f3900f8b4e538c39156bed), please check

**Comment by legalsylvain** [comment]
Hi @dalonsod. Thanks, I didn't saw it.

I made a little PR with some improvment, specially because i'll port purchase_triple_discount.

@lmiguens-solvos : could you review ?

regards.

**Comment by dalonsod** [comment]
@legalsylvain added your changes, locally tested, everything seems ok. Could you check?

**Review by legalsylvain — APPROVED** [review]
Hi @lmiguens-solvos. That's all ok for me. Thanks for having merged my PR.

- Code review and test OK.
- I Just did the migration of ``purchase_triple_discount`` ([here](https://github.com/OCA/purchase-workflow/pull/733)) and all seems OK

thanks :+1:

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
The code for updating stock.move.price_unit is not updated. This is a workaround while https://github.com/odoo/odoo/pull/29983 is not merged, but we need it right now. You can put a comment in the code referencing this Odoo pull request for removing the code when it eventually gets merged.

**Comment by pedrobaeza** [comment]
I have squashed a bit commit history.

**Comment by legalsylvain** [comment]
> The code for updating stock.move.price_unit is not updated. This is a workaround while [odoo/odoo#29983](https://github.com/odoo/odoo/pull/29983) is not merged, but we need it right now. You can put a comment in the code referencing this Odoo pull request for removing the code when it eventually gets merged.

I'm not sure to understand. your request is just to add a comment [here](https://github.com/OCA/purchase-workflow/pull/731/files?file-filters%5B%5D=.py#diff-44c8b65e17f135acc0912d72d4e4767dR10) ? or do you want other changes ?

Regards.

**Comment by saxomanu** [comment]
There is a little bug : in v12 no need to transform date in string.
In purchase_order line 86 : 
`            if self.order_id.date_order:
                date = fields.Date.to_string(
                    fields.Date.from_string(self.order_id.date_order))`

Have a traceback when use date
  File "purchase_discount/models/purchase_order.py", line 90, in _onchange_quantity
    date=date, uom_id=self.product_uom)
  File "odoo/addons/product/models/product.py", line 582, in _select_seller
    if seller.date_end and seller.date_end < date:
TypeError: unorderable types: datetime.date() < str()

For me 
`            if self.order_id.date_order:
                date = self.order_id.date_order`

**Comment by pedrobaeza** [comment]
I'm touching now for other things, so I will change that. Thanks for reporting @saxomanu

**Review by pedrobaeza — APPROVED** [review]
* Adjusted purchase.report to new inheritance structure
* Split tests in 2 files
* Added comments about PR for using the hook
* Fix date handling

**Comment by legalsylvain** [comment]
Thanks a lot @pedrobaeza for the last changes.

@ all : you can now review also ``purchase_triple_discount`` https://github.com/OCA/purchase-workflow/pull/733

kind regards.

**Pull request merged**